### DISCUSSION TITLE
What would be the most optimal way of reseting sort and filters to default (or to none then)

### DISCUSSION THREAD

**Discussion by egertaia** [TanStack/table#2352] [General]
The only way I imagine this working right now is by looping over all columns and then check if they do have filters/sort or not and then setting it to default state.

There used to be `setAllFilters` or something like that, but it seems that it has gone missing.

Any idea how that could be achieved?
I see that both, filter and sort implement method `resetSortBy` and `resetFilters` but they are not exposed.

**Comment by followbl** [comment]
This would be super useful - gonna continue digging in docs, but previously `setAllFiters` was great

**Reply by followbl** [reply]
@egertaia ...this was the solution for me...setAllFilters is still available, but now requires an array as the argument if you're trying to reset all filters...note is in the `useFilters` docs

`Note: You must call setAllFilters with an array, even if that array is empty. eg: setAllFilters([])`

this works for me

`setAllFilters([])`
